pkg/skaffold/schema/latest/config.go

```diff
@@ -994,6 +997,32 @@ type CustomDependencies struct {
 	Ignore []string `yaml:"ignore,omitempty"`
 }
 
+// CustomTest describes the custom test script provided by the user
+type CustomTest struct {
+	// Command is the custom command to be executed.
+	Command string `yaml:"command,omitempty"`
+
+	// Timeout sets the wait time for skaffold for the script to complete.
+	Timeout string `yaml:"timeout,omitempty"`
+
+	// Dependencies are the file dependencies that skaffold should watch for re-running the script.
+	Dependencies *CustomTestDependencies `yaml:"dependencies,omitempty"`
+}
+
+// CustomTestDependencies is used to specify dependencies for custom test script.
+// `paths` should be specified for file watching to work as expected.
+type CustomTestDependencies struct {
+	// Command represents a custom command that skaffold executes to obtain dependencies. The output of this command *must* be a valid JSON array.
+	Command string `yaml:"command,omitempty" yamltags:"oneOf=dependency"`
+
+	// Paths should be set to the file dependencies for this artifact, so that the skaffold file watcher knows when to retest and perform file synchronization.
+	Paths []string `yaml:"paths,omitempty" yamltags:"oneOf=dependency"`
+
+	// Ignore specifies the paths that should be ignored by skaffold's file watcher. If a file exists in both `paths` and in `ignore`, it will be ignored, and will be excluded from both retest and file synchronization.
+	// Will only work in conjunction with `paths`.
+	Ignore []string `yaml:"ignore,omitempty"`
```

Comment:
any reason why `Ignore` can't filter out paths returned by the `Command` also? Seems a bit off that `Paths` can be filtered but `Command` cannot.

Reply:
This change was made to be consistent with the custom build behavior.